lib/metadata.js

```diff
@@ -183,7 +183,7 @@ Metadata.prototype.generateOidcPEM = function generateOidcPEM(kid) {
     log.info('working on key:', key);
 
     // are we working on the right key?
-    if (!key.kid === kid) {
+    if (key.kid !== kid) {
```

Comment:
Without looking at all the code. Suppose the metadata has 3 keys, is there a code path find the 'right' key?
If the 'right' key is not found, there needs to be a way to inform the developer of this fact.